benchbuild/projects/gentoo/portage_gen.py

```diff
@@ -68,31 +68,34 @@ def __repr__(self):
             return self.__str__()
 
         def __str__(self):
-            domain, _, name = self.name.partition("_")
-            package = domain + '/' + name
-
-            uchroot = uchroot_no_args()
-            uchroot = uchroot["-E", "-A", "-C", "-w", "/", "-r"]
-            uchroot = uchroot[os.path.abspath(get_path_of_container())]
-            with local.env(CONFIG_PROTECT="-*"):
-                fake_emerge = uchroot["emerge",
-                                      "--autounmask-only=y",
-                                      "--autounmask-write=y",
-                                      "--nodeps"]
-                run(fake_emerge[package])
-
-            emerge_in_chroot = uchroot["emerge",
-                                       "-p",
-                                       "--nodeps",
-                                       package]
-            _, stdout, _ = emerge_in_chroot.run()
-
-            for line in stdout.split('\n'):
-                if package in line:
-                    _, _, package_name = line.partition("/")
-                    _, name, version = package_name.partition(name)
-                    version, _, _ = version.partition(" ")
-                    return version[1:]
+            try:
+                domain, _, name = self.name.partition("_")
+                package = domain + '/' + name
+
+                uchroot = uchroot_no_args()
+                uchroot = uchroot["-E", "-A", "-C", "-w", "/", "-r"]
+                uchroot = uchroot[os.path.abspath(get_path_of_container())]
+                with local.env(CONFIG_PROTECT="-*"):
+                    fake_emerge = uchroot["emerge",
+                                          "--autounmask-only=y",
+                                          "--autounmask-write=y",
+                                          "--nodeps"]
+                    run(fake_emerge[package])
+
+                emerge_in_chroot = uchroot["emerge",
+                                           "-p",
+                                           "--nodeps",
+                                           package]
+                _, stdout, _ = emerge_in_chroot.run()
+
+                for line in stdout.split('\n'):
+                    if package in line:
+                        _, _, package_name = line.partition("/")
+                        _, name, version = package_name.partition(name)
+                        version, _, _ = version.partition(" ")
+                        return version[1:]
+            except ProcessExecutionError:
+                print("This older package might not exist any more.")
```

Comment:
Don't print, use a logger. We need to think what we do next, that means the user requested a dead project. We might need to mark it outdated in the database and ignore it from listings?